File: Projet JavaEE Spring WebServiceRest/BMFTablet/src/main/java/com/tablet/bmf/entities/Produit.java
package com.tablet.bmf.entities;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonSetter;

@Entity
@Table(name = "produits")
public class Produit implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long idProduit;
	@NotEmpty
	private String designation;
	private String description;
	private double prix;
	private String photo;
	private int quantite;
	@ManyToOne
	@JoinColumn(name = "idCategorie")
	private Categorie categorie;

	public Long getIdProduit() {
		return idProduit;
	}

	public void setIdProduit(Long idProduit) {
		this.idProduit = idProduit;
	}

	public String getDesignation() {
		return designation;
	}

	public void setDesignation(String designation) {
		this.designation = designation;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public double getPrix() {
		return prix;
	}

	public void setPrix(double prix) {
		this.prix = prix;
	}


	public String getPhoto() {
		return photo;
	}

	public void setPhoto(String photo) {
		this.photo = photo;
	}

	public int getQuantite() {
		return quantite;
	}

	public void setQuantite(int quantite) {
		this.quantite = quantite;
	}

	public Categorie getCategorie() {
		return categorie;
	}

	public void setCategorie(Categorie categorie) {
		this.categorie = categorie;
	}

	public Produit() {
		super();

	}

	public Produit(String designation, String description, double prix,  String photo, int quantite) {
		super();
		this.designation = designation;
		this.description = description;
		this.prix = prix;
		this.photo = photo;
		this.quantite = quantite;
	}

	public Produit(String designation, Categorie c, String description, double prix, String photo,
			int quantite) {

		this.designation = designation;
		this.categorie = c;
		this.description = description;
		this.prix = prix;
		this.photo = photo;
		this.quantite = quantite;
	}

}

File: Projet Angular2/my-app/src/app/model/produit.ts
//sera transdere au format JSON
import { Categorie } from "app/model/categorie";

export class Produit {

    idProduit : number;
    designation : string;
    description : string;
    prix : number;
    selected : boolean;
    photo : string;
    quantite : number;
    categorie : Categorie;

    constructor (idProduit : number = 0,
             designation : string = "?",
             description : string = "?",
             prix : number = 0,
             selected : boolean = false,
             photo : string = "?",
             quantite : number = 0,
             categorie : Categorie
             ) {
    
        this.idProduit = idProduit;
        this.designation = designation;
        this.description = description;
        this.prix = prix;
        this.selected = selected;
        this.photo = photo;
        this.quantite = quantite;
        this.categorie = categorie;
   
    }

        


}
/*
	private Long idProduit;
	private String designation;
	private String description;
	private double prix;
	private boolean selected;
	private String photo;
	private int quantite;
	private Categorie categorie;
 */